build.md

```diff
@@ -0,0 +1,139 @@
+# Amplifier Build Instructions
+
+This document provides instructions for building Amplifier components, including contracts and the `ampd` service.
+
+## Prerequisites
+
+- Rust and Cargo (rust-version = "1.81.0" currently required)
+- Docker (for contract optimization)
+- Git (for cloning and handling submodules)
+- cargo-run-script (for running optimization scripts)
+
+## Initial Setup
+
+When cloning the repository, make sure to initialize and update submodules:
+
+```bash
+# Clone the repository
+git clone https://github.com/axelarnetwork/axelar-amplifier.git
+cd axelar-amplifier
+
+# Initialize and update submodules
+git submodule init
+git submodule update
+```
+
+If you've already cloned the repository without initializing submodules and encounter errors like:
+
+```
+Error: Custom { kind: Other, error: "protoc failed: Could not make proto path relative: proto-files/ampd/v1/ampd.proto: No such file or directory" }
+```
+
+Run the submodule initialization commands as shown above.
+
+## Building the Project
+
+### Building Everything
+
+To build the entire project:
+
+```bash
+cargo build
```

Comment:
This isn't very clear. To build a contract, you need to build for the wasm target with appropriate flags. Need to clarify that building the contract should be done via the optimize script. But building or testing a project for development purposes can be done via cargo build/test